plugins/catalog-backend/src/schema/openapi.generated.ts

```diff
@@ -1323,3 +1351,4 @@ export default {
     },
   },
 } as const;
+export default createRouter<typeof spec>(spec);
```

Comment:
This is currently set up to just create a single router for the lifetime of a plugin. Multiple calls to `createRouter` will create additional handlers on top of older plugins, which makes this a little weird. Are people likely to do that though? ie using a plugin router in more than 1 place.

Reply:
In jest tests, this can be avoided with a mock like (I would add this to the README)
```
jest.mock('../schema/openapi.generated', () => {
  const { spec } = jest.requireActual('../schema/openapi.generated');
  const { createRouter: createOpenApiRouter } = jest.requireActual(
    '@backstage/backend-openapi-utils',
  );
  return {
    __esModule: true,
    default: (createOpenApiRouter as <T>(t: T) => any)<typeof spec>(spec),
  };
});
```

Reply:
Would wrapping it up with one extra level help out? i.e. `() => createRouter<typeof spec>(spec)`.

Especially for tests I would really prefer if mocking wasn't needed for them to behave correctly.

Reply:
Yup, that's what I had initially, but liked the convenience of import router from "../schema/openapi.generated". Will move back towards that and probably make it async for other use cases.